kalite/distributed/tests/browser_tests/quiz.py

```diff
@@ -53,3 +56,16 @@ def test_quiz_first_answer_correct_not_registered(self):
         self.browser.execute_script("quizlog.add_response_log_item({correct: true});")
         self.assertEqual(self.browser.execute_script("return quizlog.get('total_correct')"), 1)
         self.assertEqual(self.browser.execute_script("return quizlog._response_log_cache[0]"), 1)
+
+
+    def test_unauthorized_request_redirect_to_login(self):
+
+        self.browse_to(
+            self.live_server_url +
+            reverse("view_playlist", kwargs={"playlist_id": PLAYLIST_ID}))
+        hash_value = urllib.urlparse(self.browser.current_url).fragment
+        self.browser.delete_all_cookies()
+        self.browser.find_element_by_id('solutionarea').find_element_by_css_selector('input[type=text]').click()
+        self.browser_send_keys(unicode("Anurag"))
+        self.browser_send_keys(Keys.RETURN)
+        self.assertEqual(self.browser.current_url, self.reverse('login') + "?next=" + reverse("view_playlist", kwargs={"playlist_id": PLAYLIST_ID}) + "#" + hash_value)
```

Comment:
Use [urlencode](https://docs.python.org/2/library/urllib.html#urllib.urlencode) instead of concatenating `"?next"` directly.